typescript/phoenix-sdk/src/utils/market.ts

```diff
@@ -197,6 +208,81 @@ export function deserializeRedBlackTree<Key, Value>(
   return tree;
 }
 
+
+/**
+ * Deserializes the trader RedBlackTree to return a map of trader keys to index 
+ *
+ * @param data The trader data buffer to deserialize
+ * @param keyDeserializer The deserializer for the tree key
+ * @param valueDeserializer The deserializer for the tree value
+ */
+ export function getTraderIndex<Key, Value>(
+  data: Buffer,
+  keyDeserializer: beet.BeetArgsStruct<Key>,
+  valueDeserializer: beet.BeetArgsStruct<Value>
+): Map<Key, number> {
```

Comment:
I'd prefer if the nasty index parsing was offloaded into a helper function that's shared by both deserializedRedBlackTree and this function.

Also probably much more accurate to name this `getNodeIndices` because it applies generally to all sokoban RBTs